detection/generate_transforms.py

```diff
@@ -191,6 +201,156 @@ def generate_detection_train_transform(
+            Set True only when the original images were read by itkreader with affine_lps_to_ras=True
+        amp: whether to use half precision
+
+    Return:
+        training transform for detection
+    """
+    if amp:
+        compute_dtype = torch.float16
+    else:
+        compute_dtype = torch.float32
+
+    train_transforms = Compose(
+        [
+            LoadImaged(keys=[image_key], image_only=False, meta_key_postfix="meta_dict"),
+            EnsureChannelFirstd(keys=[image_key]),
+            EnsureTyped(keys=[image_key, box_key], dtype=torch.float32),
+            EnsureTyped(keys=[label_key], dtype=torch.long),
+            StandardizeEmptyBoxd(box_keys=[box_key], box_ref_image_keys=image_key),
+            Orientationd(keys=[image_key], axcodes="RAS"),
+            intensity_transform,
+            EnsureTyped(keys=[image_key], dtype=torch.float16),
+            ConvertBoxToStandardModed(box_keys=[box_key], mode=gt_box_mode),
+            ConvertBoxToPointsd(keys=[box_key]),
+            AffineBoxToImageCoordinated(
+                box_keys=[box_key],
+                box_ref_image_keys=image_key,
+                image_meta_key_postfix="meta_dict",
+                affine_lps_to_ras=affine_lps_to_ras,
+            ),
+            GenerateExtendedBoxMask(
+                keys=box_key,
+                image_key=image_key,
+                spatial_size=patch_size,
+                whole_box=True,
+            ),
+            RandCropByPosNegLabeld(
+                keys=[image_key],
+                label_key="mask_image",
+                spatial_size=patch_size,
+                num_samples=batch_size,
+                pos=1,
+                neg=1,
+            ),
+            RandZoomd(
+                keys=[image_key],
+                prob=0.2,
+                min_zoom=0.7,
+                max_zoom=1.4,
+                padding_mode="constant",
+                keep_size=True,
+            ),
+            RandFlipd(
+                keys=[image_key],
+                prob=0.5,
+                spatial_axis=0,
+            ),
+            RandFlipd(
+                keys=[image_key],
+                prob=0.5,
+                spatial_axis=1,
+            ),
+            RandFlipd(
+                keys=[image_key],
+                prob=0.5,
+                spatial_axis=2,
+            ),
+            RandRotate90d(
+                keys=[image_key],
+                prob=0.75,
+                max_k=3,
+                spatial_axes=(0, 1),
+            ),
+            CoordinateTransformd(
+                keys=[point_key], refer_key=image_key, mode="world_to_image", affine_lps_to_ras=affine_lps_to_ras
+            ),
+            ConvertPointsToBoxd(keys=[point_key]),
+            ClipBoxToImaged(
+                box_keys=box_key,
+                label_keys=[label_key],
+                box_ref_image_keys=image_key,
+                remove_empty=True,
+            ),
+            BoxToMaskd(
```

Comment:
I leave this `BoxToMaskd`, `RandRotated` and `MaskToBoxd` here to achieve the same result as before.
It can also be implemented by using `CoordinateTransformd` but the issue here is that `BoxToMaskd` not only convert the bbox to the mask, it will also try to consider if the label is more like a ellipse, it will try to generate a fake ellipse mask.

Reply:
According to the discussion yesterday, v1 seems to provide identical results, but v2 is more native and accumulate less errors in the processing pipeline. Am I right or wrong here @KumoLiu ?

Reply:
Yes, v1 can provide identical results but v2 will be more easy to use. But I would suggest using v1 in this version since I didn't using v2 to run the end-to-end pipeline so I can't ensure whether it will influence the performance.

Reply:
Okay I see. That makes sense to me

Reply:
LGTM. Thanks.